dsm/resource_secret.go

```diff
@@ -195,6 +207,29 @@ func resourceCreateSecret(ctx context.Context, d *schema.ResourceData, m interfa
 		endpoint = fmt.Sprintf("sys/v1/plugins/%s", string(reqfpi))
 		operation = "POST"
 	}
+	allowed_key_justifications_policy, allow_exists := d.GetOk("allowed_key_justifications_policy")
+	allowed_missing_justifications, allow_missing_reason_exists := d.GetOk("allowed_missing_justifications")
+
+	if allow_exists && allow_missing_reason_exists {
+		if allowed_key_justifications_policy != nil && allowed_missing_justifications != nil {
+			plugin_object["google_access_reason_policy"] = map[string]interface{}{
+				"allow":                allowed_key_justifications_policy,
+				"allow_missing_reason": allowed_missing_justifications,
+			}
+		}
+	} else if allow_exists {
+		if allowed_key_justifications_policy != nil {
+			plugin_object["google_access_reason_policy"] = map[string]interface{}{
+				"allow": allowed_key_justifications_policy,
+			}
+		}
+	} else if allow_missing_reason_exists {
+		if allowed_missing_justifications != nil {
+			plugin_object["google_access_reason_policy"] = map[string]interface{}{
+				"allow_missing_reason": allowed_missing_justifications,
+			}
+		}
+	}
```

Comment:
I believe this whole section can be simplified by doing something like the following:

```go
allowed_key_justifications_policy, allow_exists := d.GetOk("allowed_key_justifications_policy")
allowed_missing_justifications, allow_missing_reason_exists := d.GetOk("allowed_missing_justifications")

policy_data := map[string]interface{}{}

if allow_exists && allowed_key_justifications_policy != nil {
    policy_data["allow"] = allowed_key_justifications_policy
}

if allow_missing_reason_exists && allowed_missing_justifications != nil {
    policy_data["allow_missing_reason"] = allowed_missing_justifications
}

// Only update if there are entries
if len(policy_data) > 0 {
    plugin_object["google_access_reason_policy] = policy_data
}
```

This would reduce the duplication of the logic and make things a little more clear to the reader. Here is a short Go playground [example](https://play.golang.com/p/Dt5vUYTwBee) (just passing some strings around).

Reply:
I'd also suggest we be consistent with `justifications` instead of also using `reason`, but this is less important than simplifying the logic above.

Reply:
Agreed with both the comments. Changed the code accordingly.